packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -185,14 +184,12 @@ class _MaterialTextSelectionControls extends TextSelectionControls {
 
   /// Builder for material-style text selection handles.
   @override
-  Widget buildHandle(BuildContext context, TextSelectionHandleType type, double textHeight) {
+  Widget buildHandle(BuildContext context, TextSelectionHandleType type, Color color, double textHeight) {
     final Widget handle = SizedBox(
       width: _kHandleSize,
       height: _kHandleSize,
       child: CustomPaint(
-        painter: _TextSelectionHandlePainter(
-          color: Theme.of(context).textSelectionHandleColor
```

Comment:
Was something about this original approach causing the color to not change when the theme was updated?  I'm trying to understand the cause of the problem and why the breaking changes are needed.  Like the parameter in EditableText.  Does passing it through there somehow make the handles rebuild when they weren't with this original code?

Reply:
The handles are inserted onto an `Overlay` as `OverlayEntry`s:
https://github.com/flutter/flutter/blob/93a3ce7a8d41ee8fb729a8fb9ffc979d9a3a75c2/packages/flutter/lib/src/widgets/text_selection.dart#L404-L409
So calling `Theme.of(context)` returns the closest `ThemeData` above the `Overlay`, which might be different from the `ThemeData` that sits above the `EditableText`, e.g.:

```               
                  /  TextSelectionHandle
Theme0 - Overlay <
                  \  Theme1 - TextField - EditableText
```
Instead of `Theme1`, currently `Theme0` is inherited by `TextSelectionHandle`.


The easy fix is to let `EditableText` take in `textSelectionHandleColor` as a parameter, and let `TextField` inherits from `Theme` (which it already does) to get the correct `ThemeData.textSelectionHandleColor` to build the text selection handles.

The fix doesn't solve the problem entirely as one of the tests demonstrated, because selection  handles don't listen to `Theme` directly ("listen to" as in an inheritor listens to an `InheritedWidget`, not sure what the correct term is).

Reply:
Oh I see, thanks for the clear explanation!

Would it be possible to make the handles use Theme1?  Like what if you passed `this.context` from TextSelectionOverlay into _buildHandle instead of the given context and let the handles use that?  Is that a major sin in Flutter?

Reply:
Haven't tried it but I think that probably won't work. An `InheritedWidget`'s inheritors have to be its descendant widgets. Moreover if the `BuildContext` (i.e. `Element`) we passed to the handle is unmounted (e.g., replaced by a new `Element` node) at some point, the app may crash.

Reply:
Good point...  What if you wrap the handles built in `showHandles` with a Theme widget?  I guess that wouldn't update the handle color when Theme1 changes either, so that's not really a complete solution.

Just trying to brainstorm ways to solve this problem with our existing theme infrastructure to avoid having a redundant textSelectionHandleColor parameter.